diff --git a/lib.commonjs/fine-tuning/token/token.js b/lib.commonjs/fine-tuning/token/token.js
index 451d43fb8cba5fe65347796f8dc870ce7262c235..b80a2f1ce0e63043b0d5b2d4fb77ff14248ed0d3 100644
--- a/lib.commonjs/fine-tuning/token/token.js
+++ b/lib.commonjs/fine-tuning/token/token.js
@@ -76,7 +76,7 @@ async function safeDynamicImport() {
 async function calculateTokenSizeViaExe(tokenizerRootHash, datasetPath, datasetType, tokenCounterMerkleRoot, tokenCounterFileHash) {
     await initNodeModules();
     const { download } = await safeDynamicImport();
-    const executorDir = path.join(__dirname, '..', '..', '..', '..', 'binary');
+    const executorDir = path.join(__dirname, '..', '..', '..', 'binary');
     const binaryFile = path.join(executorDir, 'token_counter');
     let needDownload = false;
     try {
diff --git a/lib.commonjs/fine-tuning/zg-storage/zg-storage.js b/lib.commonjs/fine-tuning/zg-storage/zg-storage.js
index a240955f1df31f46f16f5351a8f95e0d67b06071..011c51b2245589f9e13b58dc74841428e71c7a0f 100644
--- a/lib.commonjs/fine-tuning/zg-storage/zg-storage.js
+++ b/lib.commonjs/fine-tuning/zg-storage/zg-storage.js
@@ -15,7 +15,7 @@ async function upload(privateKey, dataPath, gasPrice, maxGasPrice, config) {
         const rpcUrl = config?.rpcUrl || const_1.ZG_RPC_ENDPOINT_TESTNET;
         const indexerUrl = config?.indexerUrl || const_1.INDEXER_URL_TESTNET_TURBO;
         return new Promise((resolve, reject) => {
-            const command = path_1.default.join(__dirname, '..', '..', '..', '..', 'binary', '0g-storage-client');
+            const command = path_1.default.join(__dirname, '..', '..', '..', 'binary', '0g-storage-client');
             const args = [
                 'upload',
                 '--url',
@@ -81,7 +81,7 @@ async function download(dataPath, dataRoot, config) {
     // Use provided config or default to testnet
     const indexerUrl = config?.indexerUrl || const_1.INDEXER_URL_TESTNET_TURBO;
     return new Promise((resolve, reject) => {
-        const command = path_1.default.join(__dirname, '..', '..', '..', '..', 'binary', '0g-storage-client');
+        const command = path_1.default.join(__dirname, '..', '..', '..', 'binary', '0g-storage-client');
         const args = [
             'download',
             '--file',
